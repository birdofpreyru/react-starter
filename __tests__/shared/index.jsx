import Application from 'shared';

const { JU } = jest.requireActual('@dr.pogodin/react-utils');

let mockDevFrontendMode = false;

jest.mock('@dr.pogodin/react-utils', () => {
  const TRU = jest.requireActual('@dr.pogodin/react-utils');
  return {
    ...TRU,
    isomorphy: {
      ...TRU.isomorphy,
      isDevBuild: () => mockDevFrontendMode,
    },
  };
});

test('Matches shallow snapshot', () => {
  JU.shallowSnapshot(<Application />);
});

test('Matches shallow snapshot in dev mode', () => {
  mockDevFrontendMode = true;
  JU.shallowSnapshot(<Application />);
});
