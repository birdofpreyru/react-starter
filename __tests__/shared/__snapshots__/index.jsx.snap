// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow snapshot 1`] = `
<GlobalStateProvider>
  <Memo(Connect(MetaTags))
    description="Yet another webapp based on Dr. Pogodin's React Starter Kit"
    title="Dr. Pogodin's React Starter"
  />
  <ReactTooltip
    clickable={false}
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
  <Routes />
</GlobalStateProvider>
`;

exports[`Matches shallow snapshot in dev mode 1`] = `
<GlobalStateProvider>
  <Memo(Connect(MetaTags))
    description="Yet another webapp based on Dr. Pogodin's React Starter Kit"
    title="Dr. Pogodin's React Starter"
  />
  <ReactTooltip
    clickable={false}
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
  <Routes />
  <DevTools />
</GlobalStateProvider>
`;
